datafusion/common/src/utils/mod.rs

```diff
@@ -324,18 +324,29 @@ pub fn longest_consecutive_prefix<T: Borrow<usize>>(
 /// Wrap an array into a single element `ListArray`.
 /// For example `[1, 2, 3]` would be converted into `[[1, 2, 3]]`
 /// The field in the list array is nullable.
-pub fn array_into_list_array_nullable(arr: ArrayRef) -> ListArray {
-    array_into_list_array(arr, true)
+pub fn array_into_list_array_nullable(
+    arr: ArrayRef,
+    field_name: Option<&str>,
+) -> ListArray {
+    array_into_list_array(arr, true, field_name)
 }
 
 /// Array Utils
 
 /// Wrap an array into a single element `ListArray`.
 /// For example `[1, 2, 3]` would be converted into `[[1, 2, 3]]`
-pub fn array_into_list_array(arr: ArrayRef, nullable: bool) -> ListArray {
+pub fn array_into_list_array(
+    arr: ArrayRef,
+    nullable: bool,
+    field_name: Option<&str>,
```

Comment:
Maybe make this `Option<impl Into<String>>` to align with the type in `Field::new`?

Reply:
We can but that means we'd now have to add type annotations to everywhere we call this, including those that are going to pass `None` because they don't have field elements. I think this would add bloat that doesn't really add much value. If you feel strongly, I'll make the adjustment.

Reply:
Yeah fair point, let's keep it as is.

Reply:
Since this function is public I don't think we can change its signature in a minor release 

https://docs.rs/datafusion/latest/datafusion/common/utils/fn.array_into_list_array.html

Thus I suggest we keep the existing functions but mark them as deprecated and add a new API

Maybe we can do something `builder` style to make future modifications easier

```rust
let list_array = ListArrayWrapper::new(arr)
  .with_nullable(false)
  .with_field(field)
```

Or something -- that would also give us a way to take the FieldRef directly as well as document what the defaults are 🤔